hnn_core/gui.py

```diff
@@ -298,27 +376,53 @@ def run_button_clicked(log_out, plot_out, drive_widgets, variables, tstep,
                     weights_ampa=weights_ampa,
                     weights_nmda=weights_nmda,
                     synaptic_delays=synaptic_delays,
-                    seedcore=drive['seedcore'].value
-                )
+                    seedcore=drive['seedcore'].value)
     except Exception as e:
         with log_out:
             print(f"error in reading drives {e}")
 
     with log_out:
         log_out.clear_output()
         print("start simulation")
-        variables['dpls'] = simulate_dipole(variables['net'], tstop=tstop.value, n_trials=ntrials.value)
+        if cmd_exists(mpi_cmd.value):
+            # should further allow users to adjust cores to use
+            # with MPIBackend(n_procs=multiprocessing.cpu_count() - 1,
+            #                 mpi_cmd=mpi_cmd.value):
+            # variables['dpls'] = simulate_dipole(variables['net'],
+            #                                     tstop=tstop.value,
+            #                                     n_trials=ntrials.value)
+            variables['backend'] = MPIBackend(
+                n_procs=multiprocessing.cpu_count() - 1, mpi_cmd=mpi_cmd.value)
```

Comment:
I think this is prone to errors. You should explicitly ask the user for number of cores. The MPIBackend falls back to Joblib backend if MPI is not available or installed.